### DISCUSSION TITLE
Android remote debugging with FDB over USB not working

### DISCUSSION THREAD

**Discussion by pol2095** [airsdk/Adobe-Runtime-Support#2525] [Q&A] [answered]
Hello,

since Air 50.2.1.1, Android [Remote debugging with FDB over USB](https://help.adobe.com/en_US/air/build/WS901d38e593cd1bac7b2281cc12cd6bced97-8000.html) not working on Windows.
It work using Air 50.1.1.2.
It work on iOS.

Thanks

**Comment by ajwfrost** [comment, marked as answer]
Can I check what Android version you're using, and whether you're using any IDE (like Animate)?

I've just tried following those instructions and they worked for me with 50.2.1.1 using FDB from the same package and ADB from the Android SDK (all from Windows command line rather than using an IDE). It's a fairly old Android phone though, I think Android 9.0.

Package the Android app (debug build of the SWF) with "-listen" and then the app hangs at start-up, until you launch FDB with the connect request and type 'run'...
```
> fdb -p 7936
Adobe fdb (Flash Player Debugger) [build development]
Copyright (c) 2004-2007 Adobe, Inc. All rights reserved.
(fdb) run
Trying to connect to Player
Player connected; session starting.
Set breakpoints and then type 'continue' to resume the session.
[SWF] AIRMobileAS3App.swf - 5,684 bytes after decompression
(fdb) c
[trace] Version = AND 50,2,1,1
[trace] RESIZE
[trace] Stage = 1440x1604
[trace] Screen = 1440x2900
...
```

When comparing it with 50.1.1.2, can you try doing everything the same but for just one step e.g. compile the SWF with 50.1, package it with 50.2, and debug using 50.1 again? If that causes an issue (but the same commands with the packaging done with 50.1) then it would mean the issue is either in the runtime that's packaged in the APK, or it's in the ADT packager itself...

Might also be worth uninstalling the old app and rebooting the Android phone between each test...

thanks

**Reply by pol2095** [reply]
I used command line, no IDE.
I tried on a Pixel 7 pro and Pixel 3, the app doesn't hangs at start-up using Air 50.2.1.1 but the app hangs at start-up using Air 50.1.1.2 and earlier ?
In legacy mode, same issue.
I think, the problem is in the runtime.

**Reply by ajwfrost** [reply]
Ah... okay, thanks; I think it may be a permissions thing perhaps, there was a change we made which has had some unforeseen impacts on debugging. I've just checked this and now I can reproduce what you see (immediate start-up)

Can you please add the `INTERNET` permissions into your application descriptor file:
```
	<android>
		<manifestAdditions><![CDATA[
			<manifest android:installLocation="auto">
				<uses-permission android:name="android.permission.INTERNET"/>
			</manifest>
		]]></manifestAdditions>
	</android>
```

This used to be included by default but was taken out as people were getting queries why their apps (that didn't need the internet) were asking users for permission to use the internet. But, we are putting this back in for any "debug" packages, because of issues like this :-(

**Reply by pol2095** [reply]
Thanks `<uses-permission android:name="android.permission.INTERNET"/>` solve the problem.

**Reply by teotigraphix** [reply]
>  But, we are putting this back in for any "debug" packages, because of issues like this :-(

 @ajwfrost  That's not bad, this was about the Play Store, so `debug` makes sense to just do it because of issues like this.
